taos-optin/src/tmq/raw.rs

```diff
@@ -329,21 +313,82 @@ pub(super) mod tmq {
 
         pub fn close(&mut self) {
             unsafe {
-                (self.tmq.tmq_consumer_close)(self.as_ptr());
+                (self.tmq_api.tmq_consumer_close)(self.as_ptr());
             }
         }
+
+        pub(crate) fn spawn_thread(&mut self) {
+            tracing::debug!("Spawn thread to call C function `tmq_consumer_poll`");
+
+            if self.receiver.is_none() {
+                return;
+            }
+
+            let mut receiver = self.receiver.take().unwrap();
+            let safe_tmq = SafeTmqT(self.as_ptr());
+            let tmq_api = self.tmq_api;
+            let api = self.api.clone();
+            let timeout = self.timeout;
+
+            std::thread::spawn(move || {
+                futures::executor::block_on(async {
+                    let safe_tmq = safe_tmq;
+                    let mut cache: Option<RawRes> = None;
+                    while let Some(sender) = receiver.recv().await {
```

Comment:
Only receiver.recv() is async method here, please use blocking method instead, such as flume::Receiver::recv()